src/main/java/de/hpi/bpt/argos/storage/DatabaseAccessImpl.java

```diff
@@ -0,0 +1,235 @@
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean establishConnection() {
+		try {
+			Configuration configuration = new Configuration();
+			configuration.configure();
+
+			sessionBatchSize = Integer.parseInt(configuration.getProperty("hibernate.jdbc.batch_size"));
+
+			configuration.setProperty("hibernate.connection.username", DatabaseAccess.getConnectionUsername());
+			configuration.setProperty("hibernate.connection.password", DatabaseAccess.getConnectionPassword());
+
+			if (Argos.isInTestMode()) {
+				configuration.setProperty("hibernate.connection.url",
+						String.format("jdbc:mysql://%1$s/argosbackend_test?createDatabaseIfNotExist=true", DatabaseAccess.getConnectionHost()));
+
+				// drop existing schema and re-create
+				configuration.setProperty("hibernate.hbm2ddl.auto", "create");
+			} else {
+				configuration.setProperty("hibernate.connection.url",
+						String.format("jdbc:mysql://%1$s/argosbackend?createDatabaseIfNotExist=true", DatabaseAccess.getConnectionHost()));
+
+				configuration.setProperty("hibernate.hbm2ddl.auto", "update");
+			}
+
+			sessionFactory = configuration.buildSessionFactory();
+		} catch (ServiceException e) {
+			LoggerUtilImpl.getInstance().error(logger, "cannot connect to database server", e);
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public SessionFactory getSessionFactory() {
+		return sessionFactory;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean saveArtifacts(PersistenceArtifact... artifacts) {
+		Transaction transaction = null;
+		try (Session session = sessionFactory.openSession()) {
+			transaction = session.beginTransaction();
+			int batchCount = 0;
+
+			for (PersistenceArtifact artifact : artifacts) {
+				session.saveOrUpdate(artifact);
+
+				if (++batchCount % sessionBatchSize == 0) {
+
+					logger.debug(String.format("saving artifacts... %1$d / %2$d", batchCount, artifacts.length));
+
+					session.flush();
+					session.clear();
+				}
+			}
+
+			transaction.commit();
+		} catch (Exception e) {
+			if (transaction != null) {
+				transaction.rollback();
+			}
+
+			LoggerUtilImpl.getInstance().error(logger, "cannot save/update artifacts in database", e);
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean deleteArtifacts(PersistenceArtifact... artifacts) {
+		Transaction transaction = null;
+		try (Session session = sessionFactory.openSession()) {
+			transaction = session.beginTransaction();
+
+			for (PersistenceArtifact artifact : artifacts) {
+				session.delete(artifact);
+			}
+
+			transaction.commit();
+		} catch (Exception e) {
+			if (transaction != null) {
+				transaction.rollback();
+			}
+
+			LoggerUtilImpl.getInstance().error(logger, "cannot delete artifacts in database", e);
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public <ResultType, QueryType> ResultType getArtifacts(Session session,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00119)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00119)